src/client/resources/conversions.ts

```diff
@@ -217,10 +217,11 @@ export function updateMigPlanFromValues(migPlan: any, planValues: any) {
     namespace: migPlan.metadata.namespace,
   };
 
-  if(updatedSpec.persistentVolumes) {
+  if (updatedSpec.persistentVolumes) {
     updatedSpec.persistentVolumes = updatedSpec.persistentVolumes.map(v => {
       const userPv = planValues.persistentVolumes.find(upv => upv.name === v.name);
-      v.action = userPv.type;
+      v.selection.action = userPv.type;
+      v.selection.storageClass = userPv.storageClass;
```

Comment:
storageClass is updated on the migPlan when the updatePlan is run at the end of the wizard.